src/viking.js

```diff
@@ -1,14 +1,77 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength){
+    this.health = health;
+    this.strength = strength;
+  }
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(damage){
+    this.health -= damage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength){
+    super(health, strength);
+    this.name = name;
+  }
+
+  receiveDamage(damage){
+    this.health -= damage;
+    if(this.health > 0) {
+      let message = `${this.name} has received ${damage} points of damage`
+      return message;
+    }else {
+      let message = `${this.name} has died in act of combat`
+      return message;
+    }
+  }
+
+  battleCry() {
+    return 'Odin Owns You All!'
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  receiveDamage(damage){
+    this.health -= damage;
+    if(this.health > 0) {
+      let message = `A Saxon has received ${damage} points of damage`
+      return message;
+    }else {
+      let message = `A Saxon has died in combat`
+      return message;
+    }
+  }
+}
 
 // War
-class War {}
+class War {
+  constructor() {
+    this.vikingArmy = [];
+    this.saxonArmy = []; 
+  }
+
+  addViking(viking){
+    this.vikingArmy.push(viking)
+  }
+
+  addSaxon(saxon){
+    this.saxonArmy.push(saxon)
+  }
+
+  /*vikingAttack(){
```

Comment:
la funcion de saxon al ataque 

`  saxonAttack() {
    let randomVikingIndex = Math.floor(Math.random() * this.vikingArmy.length);
    let randomViking = this.vikingArmy[randomVikingIndex];
    let randomSaxonIndex = Math.floor(Math.random() * this.saxonArmy.length);
    let randomSaxon = this.saxonArmy[randomSaxonIndex];

    let thingToReturnAtTheEnd = randomViking.receiveDamage(
      randomSaxon.attack()
    );
    if (randomViking.health <= 0) {
      this.vikingArmy.splice(randomVikingIndex, 1);
    }

    return thingToReturnAtTheEnd;
  }
`